### ISSUE TITLE
Support other build targets

### ISSUE THREAD

**Issue by EugeneKay** [Libresonic/libresonic#65]
Historically only the WAR target has been built, because that is what I maintain for myself. Other targets(Windows EXE, OS X PKG, Debian/Ubuntu Deb, RedHat/Fedora RPM, Standalone JAR booter) would be good to have, but there is a lot of work that is needed before they can become official:

- [ ] Package names must be changed to avoid potential conflicts(#52 is a start)
- [ ] Documentation needs to be created for building against each OS
- [ ] Automated build environment

**Comment by fxthomas** [comment]
Can I volunteer for this? I'd love to help setting up CI (Travis?), and I could even document the building process in the meantime.

**Comment by EugeneKay** [comment]
Yes, please feel free. @zifnab06 had a Jenkins instance up at one point, but I don't know if that is current.

**Comment by fxthomas** [comment]
Thanks! For now I'm setting up Travis on my own repo, I'll let you know once it's ready. I'm actually not familiar with Jenkins, but the build process should be easy to port once I get it up and running.

**Comment by fxthomas** [comment]
Got a working version here : https://travis-ci.org/fxthomas/libresonic (see [this commit](https://github.com/fxthomas/libresonic/commit/780c5512ae2c9bee63ce30381ebd923fc623df76)).

It's still weird for me that Maven can't pick up artifacts from inter-module dependencies automatically without installing, but other than that it appears to work for :

* Debian packages
* WAR package
* Standalone archive
* Source archive

Things that need progress :

* I had to add dependencies to `ignoredUnusedDeclaredDependencies`, not sure why I need this yet but I will find out.
* RPM packages can't be generated on Travis because `rpmlint` is only available on Xenial
* Debian packages don't build on Xenial at the moment because `lintian` is complaining a lot more than on Precise
* Windows packages need Windows
* Mac packages need a Mac

For the first 3 points it's only a matter of time, but the last two are quite annoying to setup on a CI server.

**Comment by EugeneKay** [comment]
I have decided to put this off until after 6.2, due to the Spring Boot migration. We'll revisit this around 6.3.

**Comment by trwnh** [comment]
For what it's worth, having the standalone JAR only would suffice, as it can be run with scripts on any platform. I have always used the JAR from Subsonic for this very reason, as it requires the least dependencies. That would be more useful/important than exe/pkg/deb/rpm.

**Comment by josh4trunks** [comment]
If we get a standalone jar for libresonic, I would gladly submit a port based on www/subsonic-standalone to the FreeBSD ports repository.

**Comment by muff1nman** [comment]
@josh4trunks The current pre-release artifacts for 6.2 can be used standalone. Its a little counter intuitive because they are war files, but they are also standalone executables.

**Comment by josh4trunks** [comment]
@muff1nman ohh, awesome. Are the startup flags documented somewhere. Sorry, I'm being lazy asking here.

**Comment by muff1nman** [comment]
https://github.com/Libresonic/libresonic/blob/develop/documentation/CONFIGURATION.md however that doc isnt really conclusive/complete yet. There are also the parameters to control the initial default music folder, playlist etc that would probably be helpful as a package maintainer. In addition there are some spring boot parameters that can be used as well.

**Comment by josh4trunks** [comment]
Ok, can I assume the parameters are the same as the subsonic standalone ones?

Here's what we set for FreeBSD
https://github.com/freebsd/freebsd-ports/blob/master/www/subsonic-standalone/files/subsonic.in#L76-L85

**Comment by muff1nman** [comment]
```
	-Dsubsonic.defaultMusicFolder=${subsonic_home}/music \
	-Dsubsonic.defaultPodcastFolder=${subsonic_home}/Podcast \
	-Dsubsonic.defaultPlaylistFolder=${subsonic_home}/playlists \
```
Is still relevant, except its now libresonic.X

```
	-Dsubsonic.host=${subsonic_host} \
	-Dsubsonic.port=${subsonic_port} \
```
Are no longer controlled directly by libresonic, but can be done via spring boot config:
```
-Dserver.port=4040
-Dserver.address=127.0.0.1
```
Here is the full reference on all settings for spring boot. Not all of them will apply, but definitely take a look at the `# EMBEDDED SERVER CONFIGURATION` section:

https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#common-application-properties

**Comment by josh4trunks** [comment]
Awesome, that should give me everything I need. I'll try to get this in the FreeBSD ports tree once 6.2 is released.

**Comment by EugeneKay** [comment]
This is being considered as a Goal for v6.3 in #373.

**Comment by jooola** [comment]
@fxthomas Any news on this issue ? Are you still working on this ?

**Comment by fxthomas** [comment]
I am not planning on working on this right now (not much time to contribute at the moment unfortunately), so feel free to pick it up! Thanks!

**Comment by anarcat** [comment]
What's the status here? SpringBoot (#142) seems to be closed, should it be checked off? What's the current .war build infra? Automated?

Is travis hooked up in here?
